### PR TITLE
[GEP-26] Serve workload identity discovery documents

### PR DESCRIPTION
**Pull request by vpnachev**
**What this PR does / why we need it**:
Serve workload identity discovery documents

/area security ipcei
/kind enhancement
/label ipcei/workload-identity

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/9586

**Special notes for your reviewer**:

**Release note**:

```feature operator
Gardener discovery server now can serve the OIDC discovery documents of Gardener Workload Identity. 
The openid configuration and jwks content is configured via the flags `--workload-identity-openid-configuration-file` and `--workload-identity-jwks-file` respectively, the documents are served on the paths `/garden/workload-identity/issuer/.well-known/openid-configuration` and `/garden/workload-identity/issuer/jwks` . The feature is disabled when none of the flags is set. 
```

### CODE REVIEW

File: internal/handler/workloadidentity/handler.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+func (h *Handler) HandleJWKS(w http.ResponseWriter, r *http.Request) {
+	handleRequest(h.log.WithName("jwks"), h.jwks, w, r)
+}
+
+func handleRequest(logger logr.Logger, responseData []byte, w http.ResponseWriter, r *http.Request) {
+	if w.Header().Get("Strict-Transport-Security") == "" {
+		w.Header().Set("Strict-Transport-Security", "max-age=31536000")
+	}
+
+	if isValid := handleInvalidRequest(logger.WithName("invalid-request-handler"), w, r); !isValid {
```
Comment (dimityrmirchev) [inline internal/handler/workloadidentity/handler.go:97]: I think that this named handler is too concrete and is implementation detail related. I would rather enhance the error log so we know where the error originates from.

The common functions of both `workloadidentity` and `openidmeta` handlers can be extracted as middlewares in the `handlers` package. It is not necessary that this is done with this PR. We can do it in a separate PR if you want.

```go
const (
	headerContentType = "Content-Type"
	mimeAppJSON       = "application/json"
)

func SetHSTS(h http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		if w.Header().Get("Strict-Transport-Security") == "" {
			w.Header().Set("Strict-Transport-Security", "max-age=31536000")
		}
		h.ServeHTTP(w, r)
	})
}

func AllowMethods(h http.Handler, log logr.Logger, allowedMethods ...string) http.Handler {
	var (
		responseMethodNotAllowed = []byte(`{"code":405,"message":"method not allowed"}`)
		methods                  = map[string]struct{}{}
	)
	for _, m := range allowedMethods {
		methods[m] = struct{}{}
	}
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		if _, ok := methods[r.Method]; !ok {
			w.Header().Set(headerContentType, mimeAppJSON)
			w.WriteHeader(http.StatusMethodNotAllowed)
			if _, err := w.Write([]byte(responseMethodNotAllowed)); err != nil {
				log.Error(err, "Failed writing response")
				return
			}
			return
		}
		h.ServeHTTP(w, r)
	})
}

func HandleNotFound(log logr.Logger) http.Handler {
	var responseNotFound = []byte(`{"code":404,"message":"not found"}`)
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set(headerContentType, mimeAppJSON)
		w.WriteHeader(http.StatusNotFound)
		if _, err := w.Write([]byte(responseNotFound)); err != nil {
			log.Error(err, "Failed writing response")
			return
		}
	})
}

```

**Review by dimityrmirchev — COMMENTED** [review]
Just a few nits and suggestions. Other looks good

File: internal/handler/workloadidentity/handler.go (hunk 1 above)
Comment (vpnachev) [reply internal/handler/workloadidentity/handler.go]: With 3ce6688de37abb6be02e774dc1a37eb76df001ff workloadidentity and openidmeta handlers are reusing as much as possible middleware handlers.

**Review by dimityrmirchev — APPROVED** [review]
/lgtm
/approve
